### PR TITLE
Add some tests to verify that we handle learner code running forever.

### PR DESCRIPTION
**Pull request by eyurko**
Note that these tests take about 3 seconds to run, because of how the timeout is set.

### CODE REVIEW

File: client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js [hunk 1]
```diff
@@ -31,6 +31,37 @@ describe('CodeRunnerDispatcherService', function() {
       };
       expect(errorFunction).toThrowError(Error);
     });
+
+    it('should throw a timeout error if passed infinitely looping code',
+      function() {
+        var code =
+        `while True:
```
Comment (seanlip) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: Maybe use [..., ...].join('') format, for consistency?

File: client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js [hunk 2]
```diff
@@ -31,6 +31,37 @@ describe('CodeRunnerDispatcherService', function() {
...
+          print 'a'
+        `;
+
+        CodeRunnerDispatcherService.runCodeAsync("python", code).then(
+          function(rawCodeEvalResult) {
+            expect(rawCodeEvalResult.getErrorString()).toBe(
+              'TimeLimitError: Program exceeded run time limit. on line 2')
```
Comment (seanlip) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: Why is there a period in the middle of the sentence? 

Edit: on thinking about it more, it seems like the "on line 2" part of this doesn't make sense. Perhaps we should remove that, in this case?

File: client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js [hunk 3]
```diff
@@ -31,6 +31,37 @@ describe('CodeRunnerDispatcherService', function() {
...
+          });
+    });
+
+    it('should throw a timeout error if passed infinitely recursing code',
+      function() {
+        var code = [
+        'def forgetToIgnoreSpaces(string):',
```
Comment (seanlip) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: Indent this and the next two lines by 2.

File: client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js [hunk 4]
```diff
@@ -31,6 +31,37 @@ describe('CodeRunnerDispatcherService', function() {
...
+        `;
+
+        CodeRunnerDispatcherService.runCodeAsync("python", code).then(
+          function(rawCodeEvalResult) {
+            expect(rawCodeEvalResult.getErrorString()).toBe(
+              'TimeLimitError: Program exceeded run time limit. on line 2')
+          });
+    });
+
+    it('should throw a timeout error if passed infinitely recursing code',
```
Comment (seanlip) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: The error message here looks more like "exceeded call stack size" than timeout...

File: client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js (hunk 3 above)
Comment (eyurko) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: Done.

File: client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js (hunk 4 above)
Comment (eyurko) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: Sounds good.

File: client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js (hunk 2 above)
Comment (eyurko) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: I added a check to getErrorString.

File: client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js (hunk 1 above)
Comment (eyurko) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: Sure, sounds good.

**Comment by seanlip** [comment]
Couple of things:

- The PR seems to contain a bunch of commits that don't belong to it.
- The review comments don't seem to have been addressed in the "Files changed" diff.

**Comment by eyurko** [comment]
Yes, sorry, I keep trying to avoid bringing in extra commits but the whole

git checkout master
git pull origin master
git checkout -b new-branch

thing seems to still pull in extra commits. Any advice on how to fix that in the future?

Also the comments should be rectified, now -- accidentally ran git status and then git push and forgot to commit in the middle.

File: client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js (hunk 2 above)
Comment (seanlip) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: The expected result for this test still doesn't look right?

Comment (eyurko) [inline client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js]: Yikes, you're completely right. Sorry about that -- it's been addressed.

**Comment by seanlip** [comment]
@eyurko -- could you please redo this PR cleanly? The "Commits" tab has extra stuff.

**Comment by eyurko** [comment]
See #289.

**Pull request closed without merging**